app/fetch/fetch.go

```diff
@@ -0,0 +1,54 @@
+package fetch
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+// exported so visible to main package
+type Response struct {
+	Count int     `json:"count"`
+	Rate  float32 `json:"rate"`
+}
+
+var (
+	treesAPIURL = "https://api.ecosia.org/v1/trees/count"
+	//TreeData is exported so it can be accessed from the fetch package
+	TreeData Response
+)
+
+// NewRequest creates a http client and makes the request, it returns a response and an error
+func NewRequest() (resp *http.Response, err error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", treesAPIURL, nil)
+	if err != nil {
+		return resp, err
+	}
+	resp, err = client.Do(req)
+	return resp, err
+}
+
+// Fetch returns response from Trees API and stores it in the response variable.
+func Fetch(makeRequest func() (*http.Response, error)) (int, error) {
+	resp, err := makeRequest()
+
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
```

Comment:
There is missing 

```
defer resp.Body.Close()
```
to release resources.

Reply:
good catch! thanks :smile: